File: steem-python/vote/vote.py
from steem import Steem

def vote(username, wif, identifier, weight):
    """ Vote for a post
        :param str username: Username of the account voting
        :param str wif: posting key of the account voting
        :param str identifier: Identifier for the post to upvote Takes
                               the form ``@author/permlink``
        :param float weight: Voting weight. Range: -100.0 - +100.0. May
                             not be 0.0
        :param str account: Voter to use for voting. (Optional)

        If ``voter`` is not defines, the ``default_account`` will be taken or
        a ValueError will be raised

        .. code-block:: python

            steempy set default_account <account>
    """
    s = Steem(keys=wif,
              nodes=["https://api.steemit.com", "https://rpc.buildteam.io"])
    s.commit.vote(identifier=identifier, weight=weight, account=username)

#example :

vote("username", "posting key", "@howo/introducting-steemsnippets" , 100)

File: steemjs/vote/vote.js
var steem = require('steem');

/**
 * Casts a vote.
 * @param {String} username - username of the voter account
 * @param {String} password - password of the voter account
 * @param {String} author - Author of the post
 * @param {String} permlink - permanent link of the post to comment to. eg : https://steemit.com/programming/@howo/introducting-steemsnippets the permlink is "introducting-steemsnippets"
 * @param {int} weight - Power of the vote, can range from -10000 to 10000, 10000 equals a 100% upvote. -10000 equals a 100% flag.
 */
function vote(username, password, author, permlink, weight)
{
    var wif = steem.auth.toWif(username, password, 'posting');
    steem.broadcast.vote(wif, username, author, permlink, weight, function(err, result) {
        console.log(err, result);
    });
}

// example : give a 100% upvote to the post https://steemit.com/programming/@howo/introducting-steemsnippets
vote("username", "password", "howo", "introducting-steemsnippets", 10000);